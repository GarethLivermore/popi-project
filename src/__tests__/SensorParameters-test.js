/*** Created by Gareth on 14/11/2017. ***/
import React from 'react';
import OrderQuantities from '../components/SensorParameters';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';

describe('OrderQuantities', () => {
    it('renders without crashing', () => {
        shallow(<OrderQuantities/>)
    });

    it('Renders correctly', () =>{
        const orderQuantities = renderer.create(<OrderQuantities/>).toJSON();
        expect(orderQuantities).toMatchSnapshot();
    });

    it('Collapses data displayed on button click', function () {
        const orderQuantities = mount(<OrderQuantities/>);
        orderQuantities.find('button').simulate('click');
        expect(orderQuantities.state('show')).toEqual('no');
    });

});
