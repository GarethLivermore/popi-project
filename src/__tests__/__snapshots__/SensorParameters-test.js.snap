// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderQuantities Renders correctly 1`] = `
<div>
  <h1
    style={
      Object {
        "color": "skyblue",
      }
    }
  />
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <p />
    Alert when level exceeds: 
    <br />
    <input
      onBlur={[Function]}
      type="text"
    />
  </div>
  <button
    className="btn btn-info"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginTop": 10,
      }
    }
    type="button"
  >
    Hide
  </button>
</div>
`;
